src/pages/MapOpportunities.jsx

```diff
@@ -0,0 +1,101 @@
+import React, { useRef, useCallback, useEffect } from "react";
+import { GoogleMap, useJsApiLoader } from "@react-google-maps/api";
+
+// Define the libraries needed for Google Maps
+const libraries = ['marker']; 
+
+function MapOpportunities({ opportunities, googleMapsApiKey }) {
+  // Reference to hold the map instance.
+  const mapRef = useRef(null);
+
+  const mapContainerStyle = {
+    width: "100%",
+    height: "400px",
+  };
+
+  const zoomLevelForSingleMarker = 15;
+
+  // Load the Google Maps API and the required libraries.
+  const { isLoaded, loadError } = useJsApiLoader({
+    googleMapsApiKey: googleMapsApiKey,
+    libraries: libraries,
+  });
+
+  // Callback to handle the map load event.
+  const onLoad = useCallback((map) => {
+    mapRef.current = map; // Store the map instance in the ref.
+
+    if (opportunities.length > 0) {
+      updateMarkers(map, opportunities); // Update markers based on opportunities.
+    }
+  }, [opportunities]);
+
+  // Function to update the markers on the map based on the opportunities.
+  const updateMarkers = (map, opportunities) => {
+    // Clear previous markers from the map.
+    map.markers = map.markers || [];
+    map.markers.forEach(marker => marker.setMap(null));
+    map.markers = [];
+
+    // If no opportunities, do not add any markers.
+    if (opportunities.length === 0) {
+      return;
+    }
+
+    // If only one opportunity, center the map on it and add a single marker.
+    if (opportunities.length === 1) {
```

Comment:
The only difference between this case and when there are more opportunities is the zoom / fitting bounds. Therefore, think about ways to consolidate these two blocks. E.g., even with 1 opportunity you can use the forEach loop below. to push the markers and even call `bounds.extend`. After that you can check the length of the opportunities to determine whether to apply a manual zoom or call `fitBounds`.